src/zcbenchmarks.cpp

```diff
@@ -280,48 +280,89 @@ double benchmark_large_tx(size_t nInputs)
     return timer_stop(tv_start);
 }
 
-double benchmark_try_decrypt_notes(size_t nAddrs)
+// The two benchmarks, try_decrypt_sprout_notes and try_decrypt_sapling_notes,
+// are checking a worst-case scenarios. In both we add n keys to a wallet, 
+// create a transaction using a key not in our original list of n, and then
+// check that the transaction is not associated with any of the keys in our 
+// wallet. We call assert(...) to ensure that this is true.
+double benchmark_try_decrypt_sprout_notes(size_t nKeys)
 {
     CWallet wallet;
-    for (int i = 0; i < nAddrs; i++) {
+    for (int i = 0; i < nKeys; i++) {
         auto sk = libzcash::SproutSpendingKey::random();
         wallet.AddSproutSpendingKey(sk);
     }
 
     auto sk = libzcash::SproutSpendingKey::random();
-    auto tx = GetValidReceive(*pzcashParams, sk, 10, true);
+    auto tx = GetValidSproutReceive(*pzcashParams, sk, 10, true);
 
     struct timeval tv_start;
     timer_start(tv_start);
-    auto nd = wallet.FindMySproutNotes(tx);
+    auto noteDataMap = wallet.FindMySproutNotes(tx);
+
+    assert(noteDataMap.empty());
     return timer_stop(tv_start);
 }
 
-double benchmark_increment_note_witnesses(size_t nTxs)
+double benchmark_try_decrypt_sapling_notes(size_t nKeys)
+{
+    // Set params
+    auto consensusParams = Params().GetConsensus();
+
+    auto masterKey = GetMasterSaplingSpendingKey();
+
+    CWallet wallet;
+
+    int i;
+    for (i = 0; i < nKeys; i++) {
```

Comment:
Nit: `for (int i = 0; ...`

Reply:
Spending keys `0` through `i-1` are added the wallet and then we use `i` after this for loop to generate a spending key that has not been added to the wallet: `auto sk = masterKey.Derive(i);`